src/directLine.test.ts

```diff
@@ -243,4 +243,146 @@ describe('MockSuite', () => {
+            const normalActivity = DirectLineMock.mockActivity('normal-test');
+            const scenario = function* (): IterableIterator<Observable<unknown>> {
+                yield Observable.timer(200, scheduler);
+                yield directline.postActivity(normalActivity);
+            };
+
+            subscriptions.push(lazyConcat(scenario()).observeOn(scheduler).subscribe());
+
+            const actual: Array<DirectLineExport.Activity> = [];
+            subscriptions.push(directline.activity$.subscribe(a => actual.push(a)));
+
+            scheduler.flush();
+
+            expect(normalActivity.deliveryMode).toBeUndefined();
+            expect(actual[0].deliveryMode).toBeUndefined();
+        });
+
+        test('Setting streaming overrides passed deliveryMode "normal" in activity to "stream"', () => {
+            directline = new DirectLineExport.DirectLine({ ...services, streaming: true });
+
+            const presetActivity: DirectLineExport.Message = {
+                type: 'message',
+                from: { id: 'sender' },
+                text: 'preset',
+                deliveryMode: 'normal'
+            };
+
+            const scenario = function* (): IterableIterator<Observable<unknown>> {
+                yield Observable.timer(200, scheduler);
+                yield directline.postActivity(presetActivity);
+            };
+
+            subscriptions.push(lazyConcat(scenario()).observeOn(scheduler).subscribe());
+
+            const actual: Array<DirectLineExport.Activity> = [];
+            subscriptions.push(directline.activity$.subscribe(a => actual.push(a)));
+
+            scheduler.flush();
+
+            expect(presetActivity.deliveryMode).toStrictEqual('stream');
+            expect(actual[0].deliveryMode).toStrictEqual('stream');
+        });
+
+        test('Not setting streaming preserves passed deliveryMode "normal" in activity', () => {
+            const presetActivity: DirectLineExport.Message = {
+                type: 'message',
+                from: { id: 'sender' },
+                text: 'preset-nonstream',
+                deliveryMode: 'normal'
+            };
+
+            const scenario = function* (): IterableIterator<Observable<unknown>> {
+                yield Observable.timer(200, scheduler);
+                yield directline.postActivity(presetActivity);
+            };
+
+            subscriptions.push(lazyConcat(scenario()).observeOn(scheduler).subscribe());
+
+            const actual: Array<DirectLineExport.Activity> = [];
+            subscriptions.push(directline.activity$.subscribe(a => actual.push(a)));
+
+            scheduler.flush();
+
+            expect(presetActivity.deliveryMode).toStrictEqual('normal');
+            expect(actual[0].deliveryMode).toStrictEqual('normal');
+        });
+
+        test('Streaming + 403 post returns retry and keeps deliveryMode=stream on activity', () => {
```

Comment:
This is nice corner case.

Consider test permutation so it will test everything you tested above. 2x the power.